src/main/java/com/google/cloud/tools/appengine/api/deploy/DefaultStageStandardConfiguration.java

```diff
@@ -35,123 +39,165 @@
-    this.dockerfile = dockerfile;
-  }
-
   @Override
   @Nullable
   public Boolean getEnableQuickstart() {
     return enableQuickstart;
   }
 
-  public void setEnableQuickstart(Boolean enableQuickstart) {
-    this.enableQuickstart = enableQuickstart;
-  }
-
   @Override
   @Nullable
   public Boolean getDisableUpdateCheck() {
     return disableUpdateCheck;
   }
 
-  public void setDisableUpdateCheck(Boolean disableUpdateCheck) {
-    this.disableUpdateCheck = disableUpdateCheck;
-  }
-
   @Override
   @Nullable
   public Boolean getEnableJarSplitting() {
     return enableJarSplitting;
   }
 
-  public void setEnableJarSplitting(Boolean enableJarSplitting) {
-    this.enableJarSplitting = enableJarSplitting;
-  }
-
   @Override
   @Nullable
   public String getJarSplittingExcludes() {
     return jarSplittingExcludes;
   }
 
-  public void setJarSplittingExcludes(String jarSplittingExcludes) {
-    this.jarSplittingExcludes = jarSplittingExcludes;
-  }
-
   @Override
   @Nullable
   public String getCompileEncoding() {
     return compileEncoding;
   }
 
-  public void setCompileEncoding(String compileEncoding) {
-    this.compileEncoding = compileEncoding;
-  }
-
   @Override
   @Nullable
   public Boolean getDeleteJsps() {
     return deleteJsps;
   }
 
-  public void setDeleteJsps(Boolean deleteJsps) {
-    this.deleteJsps = deleteJsps;
-  }
-
   @Override
   @Nullable
   public Boolean getEnableJarClasses() {
     return enableJarClasses;
   }
 
-  public void setEnableJarClasses(Boolean enableJarClasses) {
-    this.enableJarClasses = enableJarClasses;
-  }
-
   @Override
   @Nullable
   public Boolean getDisableJarJsps() {
     return disableJarJsps;
   }
 
-  public void setDisableJarJsps(Boolean disableJarJsps) {
-    this.disableJarJsps = disableJarJsps;
-  }
-
   @Override
   @Nullable
   public String getRuntime() {
     return runtime;
   }
 
-  public void setRuntime(String runtime) {
-    this.runtime = runtime;
+  public static class Builder {
+
+    @Nullable private File sourceDirectory;
+    @Nullable private File stagingDirectory;
+    @Nullable private File dockerfile;
+    @Nullable private Boolean enableQuickstart;
+    @Nullable private Boolean disableUpdateCheck;
+    @Nullable private Boolean enableJarSplitting;
+    @Nullable private String jarSplittingExcludes;
+    @Nullable private String compileEncoding;
+    @Nullable private Boolean deleteJsps;
+    @Nullable private Boolean enableJarClasses;
+    @Nullable private Boolean disableJarJsps;
+    @Nullable private String runtime;
+
+    public Builder setSourceDirectory(File sourceDirectory) {
+      this.sourceDirectory = Preconditions.checkNotNull(sourceDirectory);
+      return this;
+    }
+
+    public Builder setStagingDirectory(File stagingDirectory) {
+      this.stagingDirectory = Preconditions.checkNotNull(stagingDirectory);
+      return this;
+    }
+
+    public void setDockerfile(@Nullable File dockerfile) {
```

Comment:
Why are the rest of these builder methods terminal (i.e., not returning the `Builder` instance)?

Reply:
done